pkgs/top-level/all-packages.nix

```diff
@@ -11142,113 +11142,47 @@ with pkgs;
-
-  adoptopenjdk-hotspot-bin-13 = if stdenv.isLinux
-    then callPackage adoptopenjdk-bin-13-packages-linux.jdk-hotspot {}
-    else callPackage adoptopenjdk-bin-13-packages-darwin.jdk-hotspot {};
-  adoptopenjdk-jre-hotspot-bin-13 = if stdenv.isLinux
-    then callPackage adoptopenjdk-bin-13-packages-linux.jre-hotspot {}
-    else callPackage adoptopenjdk-bin-13-packages-darwin.jre-hotspot {};
-
-  adoptopenjdk-openj9-bin-13 = if stdenv.isLinux
-    then callPackage adoptopenjdk-bin-13-packages-linux.jdk-openj9 {}
-    else callPackage adoptopenjdk-bin-13-packages-darwin.jdk-openj9 {};
-
-  adoptopenjdk-jre-openj9-bin-13 = if stdenv.isLinux
-    then callPackage adoptopenjdk-bin-13-packages-linux.jre-openj9 {}
-    else callPackage adoptopenjdk-bin-13-packages-darwin.jre-openj9 {};
-
-  adoptopenjdk-bin-11-packages-linux = import ../development/compilers/adoptopenjdk-bin/jdk11-linux.nix;
-  adoptopenjdk-bin-11-packages-darwin = import ../development/compilers/adoptopenjdk-bin/jdk11-darwin.nix;
-
-  adoptopenjdk-hotspot-bin-11 = if stdenv.isLinux
-    then callPackage adoptopenjdk-bin-11-packages-linux.jdk-hotspot {}
-    else callPackage adoptopenjdk-bin-11-packages-darwin.jdk-hotspot {};
-  adoptopenjdk-jre-hotspot-bin-11 = if stdenv.isLinux
-    then callPackage adoptopenjdk-bin-11-packages-linux.jre-hotspot {}
-    else callPackage adoptopenjdk-bin-11-packages-darwin.jre-hotspot {};
-
-  adoptopenjdk-openj9-bin-11 = if stdenv.isLinux
-    then callPackage adoptopenjdk-bin-11-packages-linux.jdk-openj9 {}
-    else callPackage adoptopenjdk-bin-11-packages-darwin.jdk-openj9 {};
-
-  adoptopenjdk-jre-openj9-bin-11 = if stdenv.isLinux
-    then callPackage adoptopenjdk-bin-11-packages-linux.jre-openj9 {}
-    else callPackage adoptopenjdk-bin-11-packages-darwin.jre-openj9 {};
-
-  adoptopenjdk-bin-8-packages-linux = import ../development/compilers/adoptopenjdk-bin/jdk8-linux.nix;
-  adoptopenjdk-bin-8-packages-darwin = import ../development/compilers/adoptopenjdk-bin/jdk8-darwin.nix;
-
-  adoptopenjdk-hotspot-bin-8 = if stdenv.isLinux
-    then callPackage adoptopenjdk-bin-8-packages-linux.jdk-hotspot {}
-    else callPackage adoptopenjdk-bin-8-packages-darwin.jdk-hotspot {};
-  adoptopenjdk-jre-hotspot-bin-8 = if stdenv.isLinux
-    then callPackage adoptopenjdk-bin-8-packages-linux.jre-hotspot {}
-    else callPackage adoptopenjdk-bin-8-packages-darwin.jre-hotspot {};
-
-  adoptopenjdk-openj9-bin-8 = if stdenv.isLinux
-    then callPackage adoptopenjdk-bin-8-packages-linux.jdk-openj9 {}
-    else callPackage adoptopenjdk-bin-8-packages-darwin.jdk-openj9 {};
-
-  adoptopenjdk-jre-openj9-bin-8 = if stdenv.isLinux
-    then callPackage adoptopenjdk-bin-8-packages-linux.jre-openj9 {}
-    else callPackage adoptopenjdk-bin-8-packages-darwin.jre-openj9 {};
+  adoptopenjdk-bin-8-packages-linux  = javaPackages.compiler.adoptopenjdk-8.package-linux;
```

Comment:
Can we move this to an extra file and inherit everything that is in it?

Reply:
What would we gain by that? This is essentially doing this with the only difference being that the property names are changed to fit the old format. An additional file seems kind of redundant to me.

Reply:
We would remove about a 100 lines from all-packages which is aleady way to big and it would be really nice to group those big chunk of entries into another file.

Reply:
Wouldn't you still have to specify all keys to inherit?

So

```
  adoptopenjdk-bin-8-packages-darwin = javaPackages.compiler.adoptopenjdk-8.package-darwin;
  adoptopenjdk-hotspot-bin-8         = javaPackages.compiler.adoptopenjdk-8.jdk-hotspot;
  adoptopenjdk-jre-hotspot-bin-8     = javaPackages.compiler.adoptopenjdk-8.jre-hotspot;
  adoptopenjdk-openj9-bin-8          = javaPackages.compiler.adoptopenjdk-8.jdk-openj9;
  adoptopenjdk-jre-openj9-bin-8      = javaPackages.compiler.adoptopenjdk-8.jre-openj9;
```

would become

```
inherit (recurseIntoAttrs (callPackage ./adoptopenjdk-packages.nix { }))
  adoptopenjdk-bin-8-packages-darwin
  adoptopenjdk-hotspot-bin-8
  adoptopenjdk-jre-hotspot-bin-8
  adoptopenjdk-openj9-bin-8
  adoptopenjdk-jre-openj9-bin-8;
```

I can't really see how that is terser especially when considering that you would still need the intermediate file where all the renaming takes places. Using `//` isn't really an option since these packages are actually used elsewhere in the record.

Reply:
This PR already removes 150 lines from all-packages.nix.